nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/UnpackContent.java

```diff
@@ -145,6 +150,15 @@
             .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
             .build();
 
+    public static final PropertyDescriptor PASSWORD = new PropertyDescriptor.Builder()
+            .name("Password")
+            .displayName("Password")
+            .description("Password used for decrypting archive entries. Supports Zip files encrypted with ZipCrypto or AES")
+            .required(false)
+            .sensitive(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
```

Comment:
Should have expression language support. Preferably on the flowfile-level.

Reply:
David and I discussed this above; he originally had included Expression Language support here and removed it at my request as across the project we do _not_ want EL parsing of sensitive properties as described above.

Reply:
@alopresto read through your notes, and they make sense on the EL.

One thing we see with government customers in a number of cases is a small shop will have little to no IT budget and will just transfer data in a password-protected zip file. Sometimes it'll be pushed to SFTP. In those cases where the password is not reused, can you think of any good way we could adapt this sort of change to work with that?

Reply:
Variables (the original approach) wouldn't solve this problem, and full EL support would only solve it if the password was in a predictable flowfile attribute, which would need to be populated by some repeatable process (likely manual). Sensitive parameters solve this at the same level of intervention as variables, and if there is a repeatable process to determine the password, it can be persisted to a parameter via an API call. 

For a high-volume approach, I think you would need some correlation process between a specific flowfile `filename` attribute and the corresponding password via some lookup, and there is no mechanism to support this currently. I am designing some more advanced key management & sensitive property management functionality (likely via controller services) for other ongoing efforts, and these may provide a referenceable model for this requirement as well.